lxd/api.go

```diff
@@ -71,7 +72,24 @@ func restServer(d *Daemon) *http.Server {
 	uiEnabled := uiPath != "" && shared.PathExists(uiPath)
 	if uiEnabled {
 		uiHTTPDir := uiHTTPDir{http.Dir(uiPath)}
-		mux.PathPrefix("/ui/").Handler(http.StripPrefix("/ui/", http.FileServer(uiHTTPDir)))
+
+		// Serve the LXD user interface.
+		uiHandler := http.StripPrefix("/ui/", http.FileServer(uiHTTPDir))
+
+		// Set security headers
+		uiHandlerWithSecurity := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Permissions-Policy", "interest-cohort=()")
+			w.Header().Set("X-Content-Type-Options", "NOSNIFF")
+			w.Header().Set("X-Frame-Options", "SAMEORIGIN")
+			w.Header().Set("X-Xss-Protection", "1; mode=block")
```

Comment:
`X-Xss-Protection` seems like a deprecated/to be avoided headers nowadays: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection

Reply:
**update**: the suggested `Content-Security-Policies` header alternative to `X-Xss-Protection` seems to cause issues with OIDC login in the UI. We'll implement CSP later after the LTS is shipped